addons/mrp/tests/test_workorder_operation.py

```diff
@@ -1758,3 +1758,34 @@ def test_merge_lot(self):
         wo2.button_start()
         self.assertEqual(wo2.qty_producing, 10)
         self.assertEqual(wo2.finished_lot_id, lot1)
+
+    def test_add_move(self):
+        """ Make a production using multi step routing. Add an additional move
+        on a specific operation and check that the produce is consumed into the
+        right workorder. """
+        self.bom_finished1.consumption = 'flexible'
+        add_product = self.env['product.product'].create({
+            'name': 'Additional',
+        })
+        mo_form = Form(self.env['mrp.production'])
+        mo_form.product_id = self.finished1
+        mo_form.bom_id = self.bom_finished1
+        mo_form.product_qty = 10.0
+        mo = mo_form.save()
+
+        mo.action_confirm()
+        mo_form = Form(mo)
+        with mo_form.move_raw_ids.new() as move:
```

Comment:
this can't be the second commit and this commit can't work on its own
here you add a move once the mo is confirmed
and you did not update the code in _workorders_create to use the additional raw move